### PR TITLE
third-party: dropbear multi-session via vfork

### PR DESCRIPTION
**Pull request by timtim-ww**
Dropbear becomes capable of accepting multiple sessions and reconnecting
Still there's and issue - dropbear stops accepting new connections after reaching certain limit - needs investigation
Big patch makes allocation of server session global structures dynamic. Added functionality - storing pointers to these structures in dropbear task resources.
#1720

### CODE REVIEW

File: src/include/kernel/task/resource/dropbear_res.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @brief Declaration of structure of addreses for dropbear session structure
+ * @date 24.07.2023
+ */
+
+#ifndef DROPBEAR_RES_H_
+#define DROPBEAR_RES_H_
+
+#include <kernel/task.h>
+
+typedef unsigned int addr_t;
```
Comment (anton-bondarev) [inline src/include/kernel/task/resource/dropbear_res.h:13]: Maybe it's better to use uintptr_t for this purpose? I mean 64 bits platform can have sizeof(void *) == 8 but sizeof(unsiged int) == 4. But I 'm not sure

**Review by anton-bondarev — APPROVED** [review]
Great!
LGTM

File: src/include/kernel/task/resource/dropbear_res.h (hunk 1 above)
Comment (timtim-ww) [reply src/include/kernel/task/resource/dropbear_res.h]: sure thing, you're absolutely right

**Comment by timtim-ww** [comment]
Corrected the pointer type and added freeing of memory allocated to session

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Pull request merged**